### PR TITLE
DJ Role only

### LINKED ISSUE #463: Let server owners set a DJ role
A **DJ role** would be used to define if a user can `s!voteskip` (#458), `s!shuffle` (#462), `s!seek` (#461), etc.

### PR DESCRIPTION
**Pull request by Doges**
Closes #463

### CODE REVIEW

**Comment by pedrofracassi** [comment]
@Doges what do you want help with?

File: src/commands/configuration/config.js [hunk 1]
```diff
@@ -92,3 +94,31 @@ class ConfigPrefix extends Command {
...
+    this.parameters = new CommandParameters(this,
+      new StringParameter({ full: true, required: false, missingError: 'errors:invalidRole' })
+    )
+  }
+
+  async run ({ t, author, channel, guildDocument }, role) {
+    const regexResult = await ROLE_REGEX.exec(role)
+    let roleId = regexResult[0]
+    const embed = new SwitchbladeEmbed(author)
+    const guildrole = await channel.guild.roles.find(r => r.id === roleId)
```
Comment (bolsomito) [inline src/commands/configuration/config.js]: you *could* also search for the role name, just sayin'

File: src/utils/PermissionUtils.js [hunk 2]
```diff
@@ -13,4 +13,12 @@ module.exports = class PermissionUtils {
       return member.roles.filter(r => r.hoist).sort((a, b) => b.position - a.position).first()
     }
   }
+
+  static async isDj (client, user, channel) {
+    const guild = channel.guild
+    const role = await client.database.guilds.get(guild)
```
Comment (bolsomito) [inline src/utils/PermissionUtils.js]: `.get(guild.id)`

File: src/utils/PermissionUtils.js [hunk 3]
```diff
@@ -13,4 +13,12 @@ module.exports = class PermissionUtils {
...
+    if (!role) return
+    const hasRole = role && role.members.has(user.id)
+    return hasRole
```
Comment (bolsomito) [inline src/utils/PermissionUtils.js]: wait, this whole thing is verifying on a guildDocument if there's a user in it..?

File: src/commands/configuration/config.js (hunk 1 above)
Comment (Doges) [reply src/commands/configuration/config.js]: we are not searching for role name

Comment (bolsomito) [reply src/commands/configuration/config.js]: bUt YoU cAn, FoR eAsE oF uSe

File: src/utils/PermissionUtils.js (hunk 3 above)
Comment (Doges) [reply src/utils/PermissionUtils.js]: se tiver um cargo na database qualquer, ele vai procurar pelo cargo, se não tiver, ele vai deixar passar

Comment (bolsomito) [reply src/utils/PermissionUtils.js]: but bro, you're using ``role``, not ``{ role }`` or ``{ djRole }``

Comment (bolsomito) [reply src/utils/PermissionUtils.js]: you're defining a variable to the guildDocument, and not to the value on that certain guildDocument

Comment (Doges) [reply src/utils/PermissionUtils.js]: O

File: src/utils/PermissionUtils.js [hunk 4]
```diff
@@ -13,4 +13,12 @@ module.exports = class PermissionUtils {
       return member.roles.filter(r => r.hoist).sort((a, b) => b.position - a.position).first()
     }
   }
+
+  static async isDj (client, user, channel) {
+    const guild = channel.guild
+    const { role } = await client.database.guilds.get(guild.id)
+    if (!role) return
+    const hasRole = role && role.members.has(user.id)
+    return hasRole
```
Comment (bolsomito) [inline src/utils/PermissionUtils.js]: by the way, that would return undefined in any way, since `{ role }` doesn't exist on the schemas: it's `{ djRole }`

Comment (bolsomito) [reply src/utils/PermissionUtils.js]: aaaaaaaaand, lines 20 to 22 can be reduced to `return djRole ? djRole.members.has(user.id) : undefined`

File: src/commands/configuration/config.js (hunk 1 above)
Comment (Doges) [reply src/commands/configuration/config.js]: > bUt YoU cAn, FoR eAsE oF uSe

eu falei isso pro pedro, ele falou que nao, acabou

**Comment by Doges** [comment]
fuck it

**Pull request closed without merging**